staging/src/github.com/kubeedge/beehive/pkg/core/context/context_factory.go

```diff
@@ -2,111 +2,334 @@ package context
 
 import (
 	gocontext "context"
+	"fmt"
 	"sync"
 	"time"
 
 	"k8s.io/klog/v2"
 
+	"github.com/kubeedge/beehive/pkg/common"
+	"github.com/kubeedge/beehive/pkg/core/channel"
 	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/beehive/pkg/core/socket"
+	"github.com/kubeedge/beehive/pkg/core/socket/config"
 )
 
-// define channel type
-const (
-	MsgCtxTypeChannel = "channel"
-)
+// BeehiveContext is global context: only use for local cache to dispatch message
+type BeehiveContext struct {
+	// module -> ModuleContext/MessageContext
+	moduleModuleContext  map[string]ModuleContext
+	moduleMessageContext map[string]MessageContext
+	moduleLock           sync.RWMutex
+
+	// group -> ModuleContext/MessageContext
+	groupModuleContext  map[string]ModuleContext
+	groupMessageContext map[string]MessageContext
+	groupLock           sync.RWMutex
+
+	ctx    gocontext.Context
+	cancel gocontext.CancelFunc
+}
 
 var (
 	// singleton
-	context *beehiveContext
-	once    sync.Once
+	beehiveContext *BeehiveContext
+	once           sync.Once
 )
 
+// GetGlobalContext get global context instance
+func GetGlobalContext() *BeehiveContext {
+	return newContext()
+}
```

Comment:
Do we need this func?

Reply:
yes, to get global variable `globalContext`